BSP/STM32L431/Makefile

```diff
@@ -0,0 +1,210 @@
+PROJECT_DIR ?= $(error PROJECT_DIR is not set)
+BUILD_DIR ?= $(error BUILD_DIR is not set)
+
+######################################
+# building variables
+######################################
+# debug build?
+DEBUG = 1
+# optimization
+OPT = -Og
+
+
+#######################################
+# paths
+#######################################
+# Build path
+BSP_DIR := .
+CMSIS_DIR := ../../CMSIS
+FREERTOS_DIR := ../../FreeRTOS-Kernel
+UTIL_DIR := ../../Util
+
+######################################
+# source
+######################################
+# C sources
+C_SOURCES :=	\
+$(BSP_DIR)/Src/*.c	\
+$(BSP_DIR)/STM32L4xx_HAL_Driver/Src/*.c	\
+$(FREERTOS_DIR)/*.c	\
+$(UTIL_DIR)/Src/*.c	\
+
+# Conditionally include $(BSP_DIR)/../Inc/ based on customBSP
+ifeq ($(BSPConfig), on)
+    C_SOURCES += $(BSP_DIR)/../Src/*.c
+endif
+
+
+# Expand wildcards to get each source file individually
+C_SOURCES := $(foreach path, $(C_SOURCES), $(wildcard $(path)))
+
+# Add specific sources
+C_SOURCES += $(FREERTOS_DIR)/portable/GCC/ARM_CM4F/port.c	\
+
+# Expand wildcards if project gives more sources
+ifdef PROJECT_C_SOURCES
+C_SOURCES += $(foreach path, $(PROJECT_C_SOURCES), $(wildcard $(path)))
+endif
+
+# If a test file is defined, take out main.c and add in the test to sources
+ifdef TEST
+C_SOURCES := $(filter-out %main.c, $(C_SOURCES)) # Filter out main.c
+C_SOURCES += $(wildcard $(TEST)) # Add test file
+endif
+
+# ASM sources
+ASM_SOURCES =  \
+$(BSP_DIR)/Src/startup_stm32l431xx.s
+
+
+#######################################
+# binaries
+#######################################
+PREFIX = arm-none-eabi-
+# The gcc compiler bin path can be either defined in make command via GCC_PATH variable (> make GCC_PATH=xxx)
+# either it can be added to the PATH environment variable.
+ifdef GCC_PATH
+CC = $(GCC_PATH)/$(PREFIX)gcc
+AS = $(GCC_PATH)/$(PREFIX)gcc -x assembler-with-cpp
+CP = $(GCC_PATH)/$(PREFIX)objcopy
+SZ = $(GCC_PATH)/$(PREFIX)size
+else
+CC = $(PREFIX)gcc
+AS = $(PREFIX)gcc -x assembler-with-cpp
+CP = $(PREFIX)objcopy
+SZ = $(PREFIX)size
+endif
+HEX = $(CP) -O ihex
+BIN = $(CP) -O binary -S
+ 
+#######################################
+# CFLAGS
+#######################################
+# cpu
+CPU = -mcpu=cortex-m4
+
+# fpu
+FPU = -mfpu=fpv4-sp-d16
+
+# float-abi
+FLOAT-ABI = -mfloat-abi=hard
+
+# mcu
+MCU = $(CPU) -mthumb $(FPU) $(FLOAT-ABI)
+
+# macros for gcc
+# AS defines
+AS_DEFS = 
+
+# C defines
+C_DEFS =  \
+-DUSE_HAL_DRIVER \
+-DSTM32L431xx
+
+# AS includes
+AS_INCLUDES = 
+
+# $(addsuffix /Inc, $(MODULES))	\
+$(OS_SRC_DIRS)
+
+# C includes
+C_INCLUDES :=  \
+$(BSP_DIR)/Inc/	\
+$(BSP_DIR)/STM32L4xx_HAL_Driver/Inc/	\
+$(BSP_DIR)/STM32L4xx_HAL_Driver/Inc/Legacy/	\
+$(CMSIS_DIR)/Device/ST/STM32L4xx/Include/      \
+$(CMSIS_DIR)/Core/Include/	\
+$(FREERTOS_DIR)/include	\
+$(FREERTOS_DIR)/portable/GCC/ARM_CM4F	\
+$(UTIL_DIR)/Inc/	\
+
+# Conditionally include $(BSP_DIR)/../Inc/ based on HAL
+ifeq ($(HAL), custom)
+    C_INCLUDES += $(BSP_DIR)/../Inc/

```

Comment:
remove the tab here; see https://stackoverflow.com/a/55113847/24721924

Reply:
I fixed this in #54 so i'm not going to fix that here, this should be merged in after makefile-merge is merged in